hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/HoodieIndexUtils.java

```diff
@@ -460,4 +473,66 @@ public static <R> HoodieRecord<R> createNewTaggedHoodieRecord(HoodieRecord<R> ol
         throw new HoodieIndexException("Unsupported record type: " + recordType);
     }
   }
+
+  /**
+   * Register a metadata index.
+   * Index definitions are stored in user-specified path or, by default, in .hoodie/.index_defs/index.json.
+   * For the first time, the index definition file will be created if not exists.
+   * For the second time, the index definition file will be updated if exists.
+   * Table Config is updated if necessary.
+   */
+  public static void register(HoodieTableMetaClient metaClient, HoodieIndexDefinition indexDefinition) {
```

Comment:
note to reviewer: this is not new logic. Just moved the util method from `HoodieSparkIndexClient` to this util class.